source/server-discovery-and-monitoring/tests/monitoring/replica_set_me_mismatch.yml

```diff
@@ -0,0 +1,201 @@
+description: "Monitoring a topology that is a replica set with a me mismatch in seeds"
+uri: "mongodb://a,b"
+phases: 
+  # phase 1
+  - 
+    responses: 
+      - 
+        - "a:27017"
+        - 
+          ok: 1
+          ismaster: true
+          setName: "rs"
+          setVersion: 1
+          primary: "c:27017"
+          hosts: 
+            - "c:27017"
+            - "d:27017"
+          minWireVersion: 0
+          maxWireVersion: 4
+    outcome: 
+      events: 
+        - 
+          topology_opening_event: 
+            topologyId: "42"
+        - 
+          topology_description_changed_event: 
+            topologyId: "42"
+            previousDescription: 
+              topologyType: "Unknown"
+              servers: []
+            newDescription: 
+              topologyType: "Unknown"
+              servers: 
+                - 
+                  address: "a:27017"
+                  arbiters: []
+                  hosts: []
+                  passives: []
+                  type: "Unknown"
+                - 
+                  address: "b:27017"
+                  arbiters: []
+                  hosts: []
+                  passives: []
+                  type: "Unknown"
+        - 
+          server_opening_event: 
+            topologyId: "42"
+            address: "a:27017"
+        - 
+          server_opening_event: 
+            topologyId: "42"
+            address: "b:27017"
+        - 
+          server_description_changed_event: 
+            topologyId: "42"
+            address: "a:27017"
+            previousDescription: 
+              address: "a:27017"
+              arbiters: []
+              hosts: []
+              passives: []
+              type: "Unknown"
+            newDescription: 
+              address: "a:27017"
+              arbiters: []
+              hosts: 
+                - "c:27017"
+                - "d:27017"
+              passives: []
+              primary: "c:27017"
+              setName: "rs"
+              type: "RSOther"
```

Comment:
Why is this RSOther? I'd expect RSPrimary. RSOther is defined:

"setName" in response, "hidden: true" or not primary,  
secondary, nor arbiter.

Reply:
Ruby driver additionally requires that the primary's idea of the primary matches the address which the driver has for it. This was added for https://jira.mongodb.org/browse/RUBY-1241 in https://github.com/mongodb/mongo-ruby-driver/commit/b35b667123710fc4c9a0550cdb105b09cc2be065. A comment in the ticket indicated this may be relevant to other drivers. Thoughts?

Reply:
Huh ... sorry, can you explain some more to me? What does this rule accomplish that is not already accomplished by this part of updateRSFromPrimary?:
```
    for each server in topologyDescription.servers:
        if server.address not in description's "hosts", "passives", or "arbiters":
            remove the server and stop monitoring it
```